sklearn/utils/_array_api.py

```diff
@@ -533,3 +537,11 @@ def _estimator_with_converted_arrays(estimator, converter):
             attribute = converter(attribute)
         setattr(new_estimator, key, attribute)
     return new_estimator
+
+
+def counter_dtype(xp, dtype):  # TODO this is an ugly hack and needs to be replaced
```

Comment:
I had to figure out the integer type of the same size for the standard scaler. There should be a better way... and probably also a better name for such a function.

Reply:
As commented elsewhere, I don't think we need such a helper function. The maximum number of samples in `X` is unrelated to the floating point precision of its values: it's perfectly possible to get a very large numpy array with low precision values.